cishell/src/main/java/org/cishell/cibridge/cishell/impl/CIShellCIBridgeAlgorithmFacade.java

```diff
@@ -337,4 +348,16 @@ public void cacheAlgorithmDefinitions() {
     Map<Algorithm, CIShellCIBridgeAlgorithmInstance> getCIShellAlgorithmCIBridgeAlgorithmMap() {
         return cishellAlgorithmCIBridgeAlgorithmMap;
     }
+
+    public ObservableEmitter<AlgorithmDefinition> getAlgorithmDefinitionAddedObservableEmitter() {
+        return algorithmDefinitionAddedObservableEmitter;
+    }
+
+    public ObservableEmitter<AlgorithmDefinition> getAlgorithmDefinitionRemovedObservableEmitter() {
```

Comment:
Ideally emitters are not available outside the class at all. If you have to do this, make them protected so they can only be used by the impl classes. This will help prevent bugs and misuse in the future.